src/azure-cli/azure/cli/command_modules/signalr/_params.py

```diff
@@ -56,3 +64,17 @@ def load_arguments(self, _):
 
     with self.argument_context('signalr cors remove') as c:
         c.argument('allowed_origins', options_list=['--allowed-origins', '-a'], nargs='*', help='space separated origins that should be allowed to make cross-origin calls (for example: http://example.com:12345). To allow all, use "*"')
+
+    # Network Rule
+    with self.argument_context('signalr network-rule update') as c:
+        c.argument('connection_name', nargs='*', help='Space-separeted list of private endpoint connection name.', required=False)
+        c.argument('public_network', arg_type=get_three_state_flag(), help='The rules for public network.', required=False)
```

Comment:
Is this parameter that want to tell user enable public network?
If that so, I would recommand rename it as `enable-public-network`

Reply:
No, not to enable public network but to set network rules for public network. You can set rules for private endpoint connections if using `--connection-name <name>` and set rules for public network if using `--public-network`.

Reply:
The type of `--public-network` is `get_three_state_flag`. So it will have 2 values: true/false to allow user to fill in. According to Azure CLI convention, we recommend to prefix the parameter with a verb. And based on your indention of this parameter, we would recommend to inspire user more explicitly intuitive what this parameter does like `enable-public-network-rule`.

Reply:
According to discussion offline, we would group the arguments for `public-network` and `private-endpoint-network` separately.